### PR TITLE
fix(ios): only include presentationControllerDidDismiss when photogallery is used

### PR DESCRIPTION
**Pull request by ewanharris**
Fixes TIMOB-28100

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-28100

See jira for test case

Not sure on whether the styling/usage here is completely right, but it compiles and linting is happy. Happy to make any adjustments as needed

### CODE REVIEW

File: iphone/Classes/MediaModule.m [hunk 1]
```diff
@@ -1820,13 +1820,15 @@ - (void)popoverPresentationControllerDidDismissPopover:(UIPopoverPresentationCon
 
 #pragma mark UIAdaptivePresentationControllerDelegate
 
+#ifdef USE_TI_MEDIAOPENPHOTOGALLERY
```
Comment (vijaysingh-axway) [inline iphone/Classes/MediaModule.m:1823]: @ewanharris I think guarding the code inside the delegate method is more relevant.
Guard it with `#if defined(USE_TI_MEDIASHOWCAMERA) || defined(USE_TI_MEDIAOPENPHOTOGALLERY) || defined(USE_TI_MEDIASTARTVIDEOEDITING)
`

File: iphone/Classes/MediaModule.m [hunk 2]
```diff
@@ -1820,13 +1820,15 @@ - (void)popoverPresentationControllerDidDismissPopover:(UIPopoverPresentationCon
...
 #if IS_SDK_IOS_13
 - (void)presentationControllerDidDismiss:(UIPresentationController *)presentationController
 {
   [self closeModalPicker:picker];
   [self sendPickerCancel];
 }
 #endif
+#endif
```
Comment (vijaysingh-axway) [inline iphone/Classes/MediaModule.m:1831]:
```suggestion
#if IS_SDK_IOS_13
- (void)presentationControllerDidDismiss:(UIPresentationController *)presentationController
{
#if defined(USE_TI_MEDIASHOWCAMERA) || defined(USE_TI_MEDIAOPENPHOTOGALLERY) || defined(USE_TI_MEDIASTARTVIDEOEDITING)
  [self closeModalPicker:picker];
  [self sendPickerCancel];
#endif
}
#endif
```

File: iphone/Classes/MediaModule.m (hunk 1 above)
Comment (ewanharris) [reply iphone/Classes/MediaModule.m]: @vijaysingh-axway fixed!

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Review by ssjsamir — APPROVED** [review]
**FR Passed**, Application no longer throws an error. Tested using the following test case: 
```
var win = Ti.UI.createWindow();
var audioPlayer = Ti.Media.createAudioPlayer();
win.open();
```

**Test environment**
```
MacOS Big Sur: 11.0 Beta 5
Xcode: 12.0 Beta 5
Java Version: 1.8.0_242
Android NDK: 21.3.6528147
Node.js: 12.18.1
""NPM":"5.0.0","CLI":"8.1.0-master.11""
iphone 11 (14.0 beta)
```

**Comment by sgtcoolguy** [comment]
manually cherry-picked over to 9_3_X after solving root issue (another missing cherry-pick)

**Pull request merged**